2-collectors/scala-stream-collector/src/main/scala/com.snowplowanalytics.snowplow.collectors/scalastream/CollectorService.scala

```diff
@@ -54,43 +59,122 @@ class CollectorServiceActor(collectorConfig: CollectorConfig,
+                              ip,
+                              request,
+                              refererURI,
+                              "/" + path1 + "/" + path2,
+                              false,
+                              spCookie
+                            )._1
+                          )
+                        }
+                      }
+                    }
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    } ~
+      post {
+        path(Segment / Segment) { (path1, path2) =>
+          cookieIfWanted(cookieName) { reqCookie =>
+            optionalHeaderValueByName("User-Agent") { userAgent =>
+              optionalHeaderValueByName("Referer") { refererURI =>
+                headerValueByName("Raw-Request-URI") { rawRequest =>
+                  hostName { host =>
+                    clientIP { ip =>
+                      requestInstance { request =>
+                        entity(as[String]) { body =>
+                          complete(
+                            responseHandler.cookie(
+                              null,
+                              body,
+                              reqCookie,
+                              userAgent,
+                              host,
+                              ip,
+                              request,
+                              refererURI,
+                              "/" + path1 + "/" + path2,
+                              false,
+                              null
+                            )._1
+                          )
+                        }
+                      }
+                    }
+                  }
+                }
+              }
+            }
+          }
+        }
+      } ~
+      get {
+        path("""ice\.png""".r | "i".r) { path =>
+          cookieIfWanted(cookieName) { reqCookie =>
+            optionalHeaderValueByName("User-Agent") { userAgent =>
+              optionalHeaderValueByName("Referer") { refererURI =>
+                headerValueByName("Raw-Request-URI") { rawRequest =>
+                  hostName { host =>
+                    clientIP { ip =>
+                      requestInstance{ request =>
                         complete(
                           responseHandler.cookie(
+                            rawRequest match {
+                              case CollectorService.QuerystringExtractor(qs) => qs
+                              case _ => ""
+                            },
                             null,
-                            body,
                             reqCookie,
                             userAgent,
                             host,
                             ip,
                             request,
                             refererURI,
-                            "/" + path1 + "/" + path2,
-                            false
+                            "/" + path,
+                            true,
```

Comment:
I didn't change this, this comes from the master mainline SP sync so I think we are stuck with it unless we want to diverge further from the mainline